engine/src/RendererGLFW_GL46.cpp

```diff
@@ -0,0 +1,166 @@
+#include "Engine/Core/Rendering/Renderer/RendererGLFW_GL46.hpp"
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+#include "Engine/Core/Debug/Log.hpp"
+#include "Engine/Core/Rendering/Window/WindowGLFW.hpp"
+#include "Engine/Core/Tools/Format.hpp"
+
+using namespace Engine::Core::Renderering;
+using namespace Engine::Core::Debug;
+using namespace Engine::Core::Tools;
+using namespace std;
+
+inline void APIENTRY GLDebugMessageCallback(GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length,
+                                            const GLchar *msg, const void *data)
+{
+    const char *_source;
+    const char *_type;
+    const char *_severity;
+
+    switch (source)
+    {
+    case GL_DEBUG_SOURCE_API:
+        _source = "API";
+        break;
+
+    case GL_DEBUG_SOURCE_WINDOW_SYSTEM:
+        _source = "WINDOW SYSTEM";
+        break;
+
+    case GL_DEBUG_SOURCE_SHADER_COMPILER:
+        _source = "SHADER COMPILER";
+        break;
+
+    case GL_DEBUG_SOURCE_THIRD_PARTY:
+        _source = "THIRD PARTY";
+        break;
+
+    case GL_DEBUG_SOURCE_APPLICATION:
+        _source = "APPLICATION";
+        break;
+
+    case GL_DEBUG_SOURCE_OTHER:
+        _source = "UNKNOWN";
+        break;
+
+    default:
+        _source = "UNKNOWN";
+        break;
+    }
+
+    switch (type)
+    {
+    case GL_DEBUG_TYPE_ERROR:
+        _type = "ERROR";
+        break;
+
+    case GL_DEBUG_TYPE_DEPRECATED_BEHAVIOR:
+        _type = "DEPRECATED BEHAVIOR";
+        break;
+
+    case GL_DEBUG_TYPE_UNDEFINED_BEHAVIOR:
+        _type = "UDEFINED BEHAVIOR";
+        break;
+
+    case GL_DEBUG_TYPE_PORTABILITY:
+        _type = "PORTABILITY";
+        break;
+
+    case GL_DEBUG_TYPE_PERFORMANCE:
+        _type = "PERFORMANCE";
+        break;
+
+    case GL_DEBUG_TYPE_OTHER:
+        _type = "OTHER";
+        break;
+
+    case GL_DEBUG_TYPE_MARKER:
+        _type = "MARKER";
+        break;
+
+    default:
+        _type = "UNKNOWN";
+        break;
+    }
+
+    switch (severity)
+    {
+    case GL_DEBUG_SEVERITY_HIGH:
+        _severity = "HIGH";
+        break;
+
+    case GL_DEBUG_SEVERITY_MEDIUM:
+        _severity = "MEDIUM";
+        break;
+
+    case GL_DEBUG_SEVERITY_LOW:
+        _severity = "LOW";
+        break;
+
+    case GL_DEBUG_SEVERITY_NOTIFICATION:
+        _severity = "NOTIFICATION";
+        break;
+
+    default:
+        _severity = "UNKNOWN";
+        break;
+    }
+
+    Log::logError(stringFormat("%d: %s of %s severity, raised from %s: %s\n", id, _type, _severity, _source, msg));
+}
+
+void RendererGLFW_GL46::initialize(const WindowCreateArg &winArg) noexcept
+{
+    Log::logInitializationStart("GLFW / OpenGL 4.6 Renderer");
+
+    // Init window
+    m_pWindow = std::make_unique<WindowGLFW>(winArg);
```

Comment:
It is not the role of the Renderer to instanciate a window. An already-built window should be given to the Renderer at its construction.

Reply:
It's a way to avoid to create windows and link it to the renderer. Render is always link to a windows and canno't work without it. This technical avoid this verbose code : 
```cpp
GLFWWindow win ([arg]);
RendererGLFW_GL46 ren (win);
```
But it's true, we can dissociate it if you want to keep it more transparent

Reply:
I will separate it. Its logic